testing/file_test/test_file.cpp

```diff
@@ -93,9 +96,53 @@ struct SplitState {
   int file_index = 0;
 };
 
+static auto ExtractFilePathFromUri(llvm::StringRef uri)
+    -> std::optional<llvm::StringRef> {
+  static constexpr llvm::StringRef FilePrefix = "file:/";
+  if (!uri.starts_with(FilePrefix)) {
+    return std::nullopt;
```

Comment:
Rather than generating the error at the callsite, had you considered using `ErrorOr` here? For example:

```suggestion
    -> ErrorOr<llvm::StringRef> {
  static constexpr llvm::StringRef FilePrefix = "file:/";
  if (!uri.starts_with(FilePrefix)) {
    return ErrorBuilder() << "uri `" << *uri << "` is not a file uri";
```


At the callsite, this should allow reducing the call to just a macro assignment (just dropping a couple LOC overall):

```
  CARBON_ASSIGN_OR_RETURN(auto file_path, ExtractFilePathFromUri(*uri));
```

(I've made a corresponding suggestion below)

Reply:
Makes sense, fixed.